### DISCUSSION TITLE
Learn coding

### DISCUSSION THREAD

**Discussion by graceogechiugwu** [community/community#45477] [GitHub Education]
I want to learn python and Dart, which on is the best for the future.

**Comment by jge162** [comment]
python is a great scripting language and easy to learn.  Meanwhile, Dart's use in mobile app development. So depending and which you prefer. I would recommend Python. you can easily transition all you projects to your Github also.

**Comment by radar-blip** [comment]
Python. 
Based on popularity, its not going away anytime soon.

**Comment by shahidulislam-cse** [comment]
Here are some resources for learning Python:

1.Codecademy's Python course: https://www.codecademy.com/learn/learn-python
2.Python.org's Beginner's Guide: https://docs.python.org/3/tutorial/index.html
3.Automate the Boring Stuff with Python: https://automatetheboringstuff.com/
4.Learn Python the Hard Way: https://learnpythonthehardway.org/
5.A Byte of Python: https://python.swaroopch.com/
6.Python for Everybody: https://www.py4e.com/

I would recommend starting with Codecademy or Python.org's Beginner's Guide and then moving on to other resources once you have a basic understanding of the language.


Here are some resources for learning Dart:

1.Dart Academy: https://dart.dev/guides/language/language-tour
2.Dart Programming Tutorial: https://www.tutorialsteacher.com/dart/dart-tutorials
3.Dart for Beginners: https://www.dartlang.org/guides/language/language-tour
4.Dart Cookbook: https://dart.dev/codelabs
5.Google's Dart Programming Language: https://www.udemy.com/course/dart-programming-language
6.Dartpad: https://dartpad.dartlang.org/


I would recommend starting with Dart Academy or Dart Programming Tutorial and then moving on to other resources once you have a basic understanding of the language. Dartpad is a good tool to start experimenting with the language and check the results in real-time.

**Reply by jge162** [reply]
very detailed. nice @shahidulislam-cse
